test/functional/users_controller_test.rb

```diff
@@ -201,6 +201,51 @@ def setup
     assert_equal User.find(user.id).bio, 'Hello, there!'
   end
 
+  test 'reject update with wrong password when ui_update is true' do
+    user = users(:bob)
+    bio = users(:bob).bio
+    UserSession.create(user)
+    post :update, params: {
+      user: {
+        bio: 'Bio updated by hacker',
+        current_password: 'wrong password',
+        ui_update: 'true'
+      }
+    }
+    assert_response :redirect
+    assert_equal User.find(user.id).bio, bio
+  end
+
+  test 'update profile with correct password when ui_update is true' do
+    user = users(:bob)
+    bio = users(:bob).bio
+    UserSession.create(user)
+    post :update, params: {
+      user: {
+        bio: 'Bio updated by user',
+        current_password: 'secretive',
+        ui_update: 'true'
+      }
+    }
+    assert_response :redirect
+    assert_equal User.find(user.id).bio, 'Bio updated by user'
+  end
+
+  test 'rejecting update profile with wrong password when ui_update is nil' do
```

Comment:
OK, what I tried to do here was to see if my concern from earlier in the discussion is really valid. I tried to write a test that attempts to get past the password requirement by passing `ui_update = nil`. If my test fails, we should probably fix it. But if it doesn't, then we should be OK and can merge this! I hope this makes sense and is more clear than just explaining my concern again. Thanks!!

Reply:
And thanks for the hard work on this! I think it's going to be GREAT 🎉

Reply:
@jywarren shouldn't the test name be allowing update profile with wrong password when ui_update is nil instead of rejecting update profile.. ?